core/contracts/Registry.sol

```diff
@@ -21,31 +22,34 @@ contract Registry is RegistryInterface, MultiRole {
         DerivativeCreator
     }
 
-    // Array of all derivatives that are approved to use the UMA Oracle.
-    address[] public registeredDerivatives;
-
     // This enum is required because a WasValid state is required to ensure that derivatives cannot be re-registered.
-    enum PointerValidity { Invalid, Valid }
+    enum DerivativeValidity { Invalid, Valid }
 
-    struct Pointer {
-        PointerValidity valid;
+    // Store all key information about a derivative.
+    struct Derivative {
+        DerivativeValidity valid;
         uint128 index;
     }
 
-    // Maps from derivative address to a pointer that refers to that registered derivative in `registeredDerivatives`.
-    mapping(address => Pointer) private derivativePointers;
-
-    // Note: this must be stored outside of `registeredDerivatives` because mappings cannot be deleted and copied
-    // like normal data. This could be stored in the Pointer struct, but storing it there would muddy the purpose
-    // of the Pointer struct and break separation of concern between referential data and data.
-    struct PartiesMap {
-        mapping(address => bool) parties;
+    struct PartyMember {
+        // Each derivative address is stored in this array.
+        address[] derivatives;
+        // The index of each derivative is mapped to it's address for constant time look up and deletion.
+        mapping(address => uint) derivativeIndex;
     }
 
-    // Maps from derivative address to the set of parties that are involved in that derivative.
-    mapping(address => PartiesMap) private derivativesToParties;
+    // Array of all derivatives that are approved to use the UMA Oracle.
+    address[] public registeredDerivatives;
+
+    // Map of derivative contracts to the associated derivative struct.
+    mapping(address => Derivative) public addressToDerivatives;
+
+    // Map each party member to their associated derivatives struct.
+    mapping(address => PartyMember) private partyMembersToDerivatives;
```

Comment:
nit: naming is a little confusing because this isn't really a mapping of party members to derivatives -- it's more of a map from a party's address to their struct. Some possible suggestions:
- `partyMap`
- `parties`

Secondary nit: I'm not a huge fan of `addressToDerivatives` because the derivative and the address are kinda the same thing. However, I think if you choose to keep `addressToDerivatives` (totally fine if you like that name), this one should probably be `addressToParties` for naming consistency.

Reply:
I agree with you on both nits. this needs to be more consistent. I changed both to follow the same pattern now.